src/registrar/tests/common.py

```diff
@@ -88,6 +95,204 @@ def send_email(self, *args, **kwargs):
+    # but append additional information to it.
+    # This is useful for that scenario and outside it for
+    # identifying if values swapped unexpectedly
+    def dummy_user(self, item_name, shorthand):
+        """Creates a dummy user object,
+        but with a shorthand and support for multiple"""
+        user = User.objects.get_or_create(
+            first_name="{} First:{}".format(item_name, shorthand),
+            last_name="{} Last:{}".format(item_name, shorthand),
+            username="{} username:{}".format(item_name, shorthand),
+        )[0]
+        return user
+
+    def dummy_contact(self, item_name, shorthand):
+        """Creates a dummy contact object"""
+        contact = Contact.objects.get_or_create(
+            first_name="{} First:{}".format(item_name, shorthand),
+            last_name="{} Last:{}".format(item_name, shorthand),
+            title="{} title:{}".format(item_name, shorthand),
+            email="{}testy@town.com".format(item_name),
+            phone="(555) 555 5555",
+        )[0]
+        return contact
+
+    def dummy_draft_domain(self, item_name):
+        """Creates a dummy draft domain object"""
+        return DraftDomain.objects.get_or_create(name="city{}.gov".format(item_name))[0]
+
+    def dummy_domain(self, item_name):
+        """Creates a dummy domain object"""
+        return Domain.objects.get_or_create(name="city{}.gov".format(item_name))[0]
+
+    def dummy_alt(self, item_name):
+        """Creates a dummy website object for alternates"""
+        return Website.objects.get_or_create(website="cityalt{}.gov".format(item_name))[
+            0
+        ]
+
+    def dummy_current(self, item_name):
+        """Creates a dummy website object for current"""
+        return Website.objects.get_or_create(website="city{}.com".format(item_name))[0]
+
+    def get_common_domain_arg_dictionary(
+        self,
+        item_name,
+        org_type="federal",
+        federal_type="executive",
+        purpose="Purpose of the site",
+    ):
+        """Generates a generic argument list for most domains"""
+        common_args = dict(
+            organization_type=org_type,
+            federal_type=federal_type,
+            purpose=purpose,
+            organization_name="{} Testorg".format(item_name),
+            address_line1="{} address 1".format(item_name),
+            address_line2="{} address 2".format(item_name),
+            is_policy_acknowledged=True,
+            state_territory="NY",
+            zipcode="10002",
+            type_of_work="e-Government",
+            anything_else="There is more",
+            authorizing_official=self.dummy_contact(item_name, "authorizing_official"),
+            submitter=self.dummy_contact(item_name, "submitter"),
+            creator=self.dummy_user(item_name, "creator"),
+        )
+        return common_args
+
+    # This can be boiled down more, though for our purposes this is OK
```

Comment:
Maybe remove this? if it's something to come back to it would be a TODO and I don't think this warrants a todo